packages/core/lib/services/credential/service.js

```diff
@@ -79,11 +81,17 @@ class CredentialService {
     for (const signedVC of credentials) {
       vp.addCredential(signedVC);
     }
-    const kp = getKeypairFromDoc(keyDoc);
-    kp.signer = kp.signer();
-    const suite = getSuiteFromKeyDoc(kp);
     vp.setHolder(keyDoc.controller);
-    return vp.sign(suite, challenge, domain);
+    keyDoc.keypair = keyDocToKeypair(keyDoc, getDock());
+    return vp.sign(
```

Comment:
why was this change required when vp.sign using suite has been working just fine afaik?

Reply:
it does not work if the key type is Sr25519